### DISCUSSION TITLE
Атозаполнение по несколько полей

### DISCUSSION THREAD

**Discussion by Leo-ui-528** [totumonline/totum-ru-issues-and-discussions#255] [Как это сделать?] [answered]
Здравствуйте, подскажите пожалуйста, как реализовать заполнение нескольких полей документа в таблице в цикле из отдельной таблице. Например есть таблица документ(1) из нескольких полей. Другая цикл, в ней документ, который состоит из полей таблицы (1). Мы заполняем одно селектом из таблицы, поле номер, например, остальные заполняются автоматически.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Два пути может быть — **тянуть** или **толкать** или комбинировано.

**Тянуть:**

Делается через Коды в тех полях, которые надо заполнить. Например мы заполняем поле с номером товара в поле `item` и нам нужно вытянуть его стоимость в поле стоимость:

```
=: select(table: 'price'; field: 'price'; where: 'id' = #item) 
```

**Толкать:**

Делается через Коды действия. По триггеру изменение мы пишем для той-же задачи:

```
=: set(table: $#ntn; field: 'price' = $price)

price: select(table: 'price'; field: 'price'; where: 'id' = $#nfv) 
```
